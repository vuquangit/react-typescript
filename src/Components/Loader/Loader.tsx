import React, { FC } from 'react'

import { LoaderWrapper, LoaderContent } from './Loader.styled'
import { BaseLoaderProps } from './Loader.types'

const Loader: FC<BaseLoaderProps> = ({ size = 24 }) => {
  return (
    <LoaderWrapper>
      <LoaderContent size={size} data-testid="loader-content">
        Loader...
      </LoaderContent>
    </LoaderWrapper>
  )
}

export default Loader
