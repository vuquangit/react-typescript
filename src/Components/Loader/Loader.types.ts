export interface ILoaderProps {
  size?: number
}

export declare type BaseLoaderProps = ILoaderProps
