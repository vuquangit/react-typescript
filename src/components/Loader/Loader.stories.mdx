import { Meta, Story, Canvas, ArgsTable } from '@storybook/addon-docs/blocks'
import { action } from '@storybook/addon-actions'
import Loader from '.'
import Space from 'stories/components/styled-system/Space.mdx'
import Styles from 'stories/styles.mdx'

<Meta title="Story/Components/Loader" component={Loader} />

<Styles />

# Loader

With `MDX` we can define a story for `Loader` right in the middle of our
Markdown documentation.

export const Template = (args) => <Loader {...args} />

<Canvas>
  <Story name="Normal" args={{}}>
    {Template.bind({})}
  </Story>
</Canvas>

<div class="subheading">Props</div>

<Space />
