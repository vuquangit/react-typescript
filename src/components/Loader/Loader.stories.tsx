import React from 'react'
import { Story, Meta } from '@storybook/react/types-6-0'
import Wrapper from '@/test/supports/AppWrapper'

import Loader from '.'
import { ILoaderProps } from './Loader.types'

export default {
  title: 'Story/Components/Loader',
  component: Loader,
  argTypes: {},
} as Meta

const Template: Story<ILoaderProps> = (args) => (
  <Wrapper themeMode="light">
    <Loader {...args} />
  </Wrapper>
)

export const Normal = Template.bind({})
Normal.args = {
  size: 24,
}
