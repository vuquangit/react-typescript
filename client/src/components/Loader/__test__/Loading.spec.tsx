/**
 *
 * Tests for Loader
 *
 */

import React from 'react'

import { render } from '@testing-library/react'
import Loader from '..'

describe('App Loader', () => {
  it('Test default Loader with no size', () => {
    const wrapper = render(<Loader />)
    expect(wrapper).toMatchSnapshot()
  })

  it('Test default Loader with size 48px', () => {
    const { container, getByTestId } = render(<Loader size={48} />)
    expect(container).toMatchSnapshot()
    expect(getByTestId('loader-content')).toHaveStyleRule('width', '48px')
    expect(getByTestId('loader-content')).toHaveStyleRule('height', '48px')
  })
})
