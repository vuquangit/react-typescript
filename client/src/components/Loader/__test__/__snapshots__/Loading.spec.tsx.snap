// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Loader Test default Loader with no size 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 100%;
  height: 100%;
}

.c1 {
  -webkit-animation: load5 1.1s infinite ease;
  animation: load5 1.1s infinite ease;
  border-radius: 50%;
  font-size: 25px;
  margin: 100px auto;
  position: relative;
  text-indent: -9999em;
  -webkit-transform: translateZ(0);
  -ms-transform: translateZ(0);
  transform: translateZ(0);
  height: 24px;
  width: 24px;
}

<body>
    <div>
      <div
        class="c0"
      >
        <div
          class="c1"
          data-testid="loader-content"
          size="24"
        >
          Loader...
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-bdfBwQ cDUTWZ"
    >
      <div
        class="sc-gsTCUz gJCNhw"
        data-testid="loader-content"
        size="24"
      >
        Loader...
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`App Loader Test default Loader with size 48px 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 100%;
  height: 100%;
}

.c1 {
  -webkit-animation: load5 1.1s infinite ease;
  animation: load5 1.1s infinite ease;
  border-radius: 50%;
  font-size: 25px;
  margin: 100px auto;
  position: relative;
  text-indent: -9999em;
  -webkit-transform: translateZ(0);
  -ms-transform: translateZ(0);
  transform: translateZ(0);
  height: 48px;
  width: 48px;
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
      data-testid="loader-content"
      size="48"
    >
      Loader...
    </div>
  </div>
</div>
`;
